services/src/main/java/org/fao/geonet/api/records/MetadataApi.java

```diff
@@ -404,6 +404,10 @@ private Object getRecordAs(
             }
         }
 
+        if (!metadataUtils.isMetadataAvailableInPortal(Integer.parseInt(mdId))) {
```

Comment:
Shouldn't this check be added to ApiUtils.canViewRecord and ApiUtils.canEditRecord so that all api's ensure that the metadata being edited or viewed exists from the current portal of the api call?

I'm also not sure if this could end up causing performance issues?

Reply:
What I mean is that it seems like if the portal is used to exclude data. In that case I would expect that the logic to be applied to all api's that are based on a metadata id. I would that the logic would be added in the canView and canEdit function so that it is checked consistently in one location.  Otherwise we are going to add this logic to all apis.

i.e. here are a couple other api's that can access data from another portal(i.e. myportal) even though they cannot locate the data using search from the portal(myportal).
I'm guessing that the list of api's that depend on metadata is larger than this.  

http://localhost:8080/geonetwork/myportal/api/records/123d4aba-93b2-4446-a376-517ddda3a75b/permalink
http://localhost:8080/geonetwork/myportal/api/records/123d4aba-93b2-4446-a376-517ddda3a75b/sharing
http://localhost:8080/geonetwork/myportal/api/records/123d4aba-93b2-4446-a376-517ddda3a75b/related?approved=true&rows=100
http://localhost:8080/geonetwork/myportal/api/records/123d4aba-93b2-4446-a376-517ddda3a75b/popularity

If you just want fix this in a one by one approach then this PR seems OK to fix the formatter issue.  But other api's should also be fixed if this is how the portals should work.

Reply:
@ianwallen I agree and can add in `ApiUtils` a method to do all these checks and reuse it for other API requests. 

But note that all API end-points use the portal in the URL path. So where to stop?

For example, all record operations `/records...` need to be updated? Batch process or reports also should take this account?

**I think the main issue is that the portal should be only used for operations related to searches or retrieval of metadata (formatters).**

All other API's, in my opinion should not have the portal stuff in the URL. But this is also a major design change, that should be done in a future GeoNetwork 5 version or so.

--- 

In summary, I would make sure that formatters and searches filter properly the metadata  for the portal. All other services as said, for me is not correct that the portal is part of that API's.